controllers/auth.js

```diff
@@ -0,0 +1,97 @@
+const bcrypt = require('bcrypt')
+const jwt = require('jwt-simple')
+const passport = require('passport')
+const passportJWT = require('passport-jwt')
+const restify = require('restify')
+
+const db = require('../services/db')
+
+const saltRounds = 10
+
+const jwtStrategyOptions = {
+  jwtFromRequest: passportJWT.ExtractJwt.fromAuthHeaderWithScheme('Bearer'),
+  secretOrKey: process.env.JWT_SECRET
+}
+
+// Create a token
+const makeToken = module.exports.makeToken = (payload) => {
+  return jwt.encode(payload, process.env.JWT_SECRET)
+}
+
+// Check user credentials and return token if valid
+module.exports.authenticate = (req, res, next) => {
+  if (req.body.email && req.body.password) {
+    return db.get('organization', 'email', req.body.email)
+      .then(organization => {
+        return bcrypt.compare(req.body.password, organization.password)
+      })
+      .then(valid => {
+        if (valid) {
+          res.send({message: 'organization credentials are valid'})
+        } else {
+          throw new restify.UnauthorizedError(
+            'email and password combination is not valid')
+        }
+      })
+      .catch(next)
+  } else {
+    return next(new restify.UnprocessableEntityError('missing required fields'))
+  }
+}
+
+// Initialize Passport middleware
+module.exports.initialize = passport.initialize()
+
+// Register an organization
+module.exports.register = (req, res, next) => {
+  if (req.body.name && req.body.email && req.body.password) {
```

Comment:
Good point. I think I might add a unique constraint to the email column in the database, ensuring that an error will be returned if an email is non-unique. What do you think?